src/nomad_parser_fhiaims/parsers/parser.py

```diff
@@ -0,0 +1,195 @@
+            ],
+            'TPSSloc gradient-corrected functionals': [
+                {'name': 'MGGA_C_TPSSLOC'},
+                {'name': 'MGGA_X_TPSS'},
+            ],
+            'hybrid B3LYP functional': [{'name': 'HYB_GGA_XC_B3LYP5'}],
+            'Hartree-Fock': [{'name': 'HF_X'}],
+            'HSE': [{'name': 'HYB_GGA_XC_HSE03'}],
+            'HSE-functional': [{'name': 'HYB_GGA_XC_HSE06'}],
+            'hybrid-PBE0 functionals': [
+                {'name': 'GGA_C_PBE'},
+                {
+                    'name': 'GGA_X_PBE',
+                    'weight': lambda x: 0.75 if x is None else 1.0 - x,
+                },
+                {'name': 'HF_X', 'weight': lambda x: 0.25 if x is None else x},
+            ],
+            'hybrid-PBEsol0 functionals': [
+                {'name': 'GGA_C_PBE_SOL'},
+                {
+                    'name': 'GGA_X_PBE_SOL',
+                    'weight': lambda x: 0.75 if x is None else 1.0 - x,
+                },
+                {'name': 'HF_X', 'weight': lambda x: 0.25 if x is None else x},
+            ],
+            'Hybrid M06 gradient-corrected functionals': [
+                {'name': 'MGGA_C_M06'},
+                {'name': 'HYB_MGGA_X_M06'},
+            ],
+            'Hybrid M06-2X gradient-corrected functionals': [
+                {'name': 'MGGA_C_M06_2X'},
+                {'name': 'HYB_MGGA_X_M06'},
+            ],
+            'Hybrid M06-HF gradient-corrected functionals': [
+                {'name': 'MGGA_C_M06_HF'},
+                {'name': 'HYB_MGGA_X_M06'},
+            ],
+            'Hybrid M08-HX gradient-corrected functionals': [
+                {'name': 'MGGA_C_M08_HX'},
+                {'name': 'HYB_MGGA_X_M08_HX'},
+            ],
+            'Hybrid M08-SO gradient-corrected functionals': [
+                {'name': 'MGGA_C_M08_SO'},
+                {'name': 'HYB_MGGA_X_M08_SO'},
+            ],
+            'Hybrid M11 gradient-corrected functionals': [
+                {'name': 'MGGA_C_M11'},
+                {'name': 'HYB_MGGA_X_M11'},
+            ],
+        }
+
+    def get_xc_functionals(self, xc: str) -> List[Dict[str, Any]]:
+        return [dict(name=functional.get('name')) for functional in self._xc_map.get(xc, [])]
+
+    def get_energies(self, source: Dict[str, Any]) -> Dict[str, Any]:
+        total_keys = ['Total energy uncorrected', 'Total energy']
+        energies = {}
+        contributions = []
+        for key, val in source.get('energy', {}).items():
+            if key in total_keys:
+                energies.setdefault('value', val)
+            else:
+                contributions.append({'name': key, 'value': val})
+        for key, val in source.get('energy_components', [{}])[-1].items():
+            contributions.append({'name': key, 'value': val})
+        energies['contributions'] = contributions
+        return energies
+
+    def get_sections(self, source: Dict[str, Any], **kwargs) -> List[Dict[str, Any]]:
```

Comment:
Is the output here meant to be passed on to other mapping annotations?

Reply:
yes, if it is a relative path.

Reply:
Same question about the `MappingAnnotation` docs.